src/components/Form.tsx

```diff
@@ -84,33 +107,51 @@ export const Form = ({
     function closeForm(): void {
       setIsFormVisible(false);
       setTimeout(() => {
-        setSelectedOption(null);
+        setSelectedJuz(undefined);
+        setSelectedApproach(undefined);
         setIsCancelConfirmationVisible(false);
       }, 500);
     }
 
+    function buildPayload(): Payload {
+      return {
+        // @ts-expect-error handled undefined value
+        juz: selectedJuz.value,
+        // @ts-expect-error handled undefined value
+        approachId: selectedApproach.value,
+      };
+    }
+
+    function isChanged(): boolean {
+      if (!selectedJuz && !selectedApproach) return false;
+      return true;
+    }
+
+    function isSaveable(): boolean {
+      if (!selectedJuz || !selectedApproach) return false;
+      return true;
+    }
```

Comment:
these functions can be converted into useMemo

Reply:
currently will mark it with some tag for tech debt